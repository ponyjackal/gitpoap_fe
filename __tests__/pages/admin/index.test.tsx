import { screen } from '@testing-library/react';
import Admin from '../../../src/pages/admin';
import { renderWithTheme } from '../../__utils__/renderWithTheme';

describe('Admin Page', () => {
  it('renders the Admin List', () => {
    const { container } = renderWithTheme(<Admin />);
    const adminPage = container.firstChild;

    const createGitpoapsItems = screen.getAllByText(/Create GitPOAPs/);
    const claimsDashboardItem = screen.getByText(/Claims Dashboard/i);
    const reposDashboardItem = screen.getByText(/Repos Dashboard/i);
    const vitalsDashboardItem = screen.getByText(/Vitals Dashboard/i);

    for (const item of createGitpoapsItems) {
      expect(item).toBeInTheDocument();
    }

    expect(claimsDashboardItem).toBeInTheDocument();
    expect(reposDashboardItem).toBeInTheDocument();
    expect(vitalsDashboardItem).toBeInTheDocument();

    expect(adminPage).toMatchSnapshot();
  });
});
