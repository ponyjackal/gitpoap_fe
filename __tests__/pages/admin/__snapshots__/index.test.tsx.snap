// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page renders the Admin List 1`] = `
.emotion-1 {
  margin: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  padding: 8px;
  margin-left: 0;
  -webkit-flex-basis: 83.33333333333334%;
  -ms-flex-preferred-size: 83.33333333333334%;
  flex-basis: 83.33333333333334%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 83.33333333333334%;
}

@media (min-width: 401px) {
  .emotion-3 {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    margin-left: 0;
  }
}

@media (min-width: 601px) {
  .emotion-3 {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    margin-left: 0;
  }
}

@media (min-width: 901px) {
  .emotion-3 {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    margin-left: 0;
  }
}

@media (min-width: 1201px) {
  .emotion-3 {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    margin-left: 0;
  }
}

@media (min-width: 1537px) {
  .emotion-3 {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    margin-left: 0;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  gap: 16px;
}

.emotion-7 {
  font-family: PT Mono,monospace;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: 0.875rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: normal;
  color: #E2E2EE;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
}

.emotion-7:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-7:focus:not(:focus-visible) {
  outline: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="flex: 1; margin-top: 2.5rem; margin-bottom: 1.25rem;"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <a
          href="/admin/codes"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Add Codes
          </div>
        </a>
        <a
          href="/admin/gitpoap/create-multiple"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Create GitPOAPs
          </div>
        </a>
        <a
          href="/admin/gitpoap/create-multiple-event"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Create GitPOAPs for Events
          </div>
        </a>
        <a
          href="/admin/gitpoap/claims"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Claims Dashboard
          </div>
        </a>
        <a
          href="/admin/repos"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Repos Dashboard
          </div>
        </a>
        <a
          href="/admin/vitals"
        >
          <div
            class="emotion-6 sc-313abcb3-0 sc-e5f22f22-0 hOflFZ ebzISv emotion-7"
          >
            Vitals Dashboard
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
